### DISCUSSION TITLE
Any guide on scaling Apache SkyWalking?

### DISCUSSION THREAD

**Discussion by TaskDead** [apache/skywalking#11878] [Q&A]
I am trying to create a method to determine the amount of hardware to be used for apache skywalking, as storage I am using opensearch
can someone help me with a documentation or any guide to do so.

**Comment by wu-sheng** [comment]
Size is determined by many things, such as
- sampling rate
- scope of your monitoring
- TTL, how long you want to keep data
- language agents, ebpf, or mesh?


This can't be wrote by a general document, you should set up your scale step by step.

**Comment by TaskDead** [comment]
for OpenSearch I have a sizing guide (30 days retention) , for scope I am looking at just java agents lets say 50  and 1-2 kubernetes cluster .
we can assume  sampling rate as 100%

**Reply by wu-sheng** [reply]
That is not enough, for example,
- How many RPS would land on your applications
- How many spans per requests you prefer to collect via different plugins and manual tracing code.

Note, it is better you could run that baseline you have, as you hold your applications deployment. We don't have that magic to get a formula.

**Reply by TaskDead** [reply]
ok , thanks will do.
